### ISSUE TITLE
Does this include vJass

### ISSUE THREAD

**Issue by grandeljay** [lep/jassdoc#121]

**Comment by lep** [comment]
No, vjass is well documented in the [jasshelpermanual](https://raw.githack.com/wurstscript/WurstScript/master/Wurstpack/vexorianjasshelper/jasshelpermanual.html) and in the documentation of any vjass library one would use. This repository is for the interface between jass (and now also lua of course) and the functionality provided by warcraft 3.
Also while not strictly neccessary, vjass is not easy to parse.

Maybe of some interest or curiosity: Barade also wrote some doxygen like tool for vjass here https://github.com/tdauth/vjassdoc
Scraping all the systems on hive and making their API searchable would be pretty cool but out of scope for this.

**Comment by grandeljay** [comment]
I see, thanks!

> Barade also wrote some doxygen like tool for vjass here https://github.com/tdauth/vjassdoc
Scraping all the systems on hive and making their API searchable would be pretty cool but out of scope for this.

I understand that vJass is out of the scope of this project. This seems to be one of the few (or only?) _still maintained_ projects regarding documentation. I find it easy to learn (v)Jass when I can simply lookup a functions description or types on the [Jassbot site](https://lep.duckdns.org/app/jassbot/). It's a big pity for me to see it "end" with Jass. There is so much more potential in my opinion.

The Jassbot has become a convenient learning haven for me, I hope you might reconsider the scope of this project!

**Comment by Luashine** [comment]
vJass, Zinc are not standalone, they turn their extended syntax code into regular Jass (a preprocessor does that, JassHelper). Zinc isn't worthy of mention, but vJass is powerful enough to need some code documentation/auto-completion on its own (I've read the entire manual to make the new notepad++ syntax highlighter and trust me, no single person has ever used all of its features). There's no use in documenting "vJass" because it's a self-contained thing.

If you meant writing documentation for the libraries (aka "systems") out there, this is asking to comment someone else's code. The plan would sound like "document all (v)Jass code that was ever written!" The author should've done that... failing that, it'd be possible to cover a few very popular libraries but why? Ripping the code from the Hive (example) to have it hosted elsewhere (Github) to enable collaboration for some code comments? I think the respective comment section should be used, though this doc generator LeP linked looks neat.

For now Ricola has a grande idea of looking through and [categorizing currently published libraries](https://www.hiveworkshop.com/threads/in-construction-classification-of-gui-jass-vjass-systems-as-of-today.350925/#post-3595358). That's the closest thing going in the direction you want.

> This seems to be one of the few (or only?) still maintained projects regarding documentation.

**The only,** though Water (with me on the side) is spear-heading the file format work. We are only a handful people total, lep's supporting the code and infrastructure. From time to time I try to pitch the idea of contributing here to mappers on Hive's Discord but they aren't interested despite encountering problems that could've been long discovered and described once and for all. I was in your shoes when I tried to learn Jass. That's why I decided to contribute to jassdoc. The language was a lesser problem than the API.

**Comment by grandeljay** [comment]
> If you meant writing documentation for the libraries (aka "systems") out there, this is asking to comment someone else's code. 

> Ripping the code from the Hive (example) to have it hosted elsewhere (Github) to enable collaboration for some code comments?

Yeah, I think that sounds good. The information is available at the hiveworkshop but it's not a standardised format, like here. I would have to actually read (probably) large amount of texts and filter out the information I need. I'd have to use my brain!
